datamol/reactions/_reactions.py

```diff
@@ -0,0 +1,109 @@
+from functools import singledispatch
+from typing import Union
+import numpy as np
+from rdkit.Chem import rdChemReactions
+from rdkit.Chem import AllChem
+
+import datamol as dm
+
+
+ATTACHING_RXN = rdChemReactions.ReactionFromSmarts("[*;h:1]>>[*:1][*]")
+
+
+def from_smarts(rxn_smarts: str):
+    """Create a reaction from smarts"""
+    return rdChemReactions.ReactionFromSmarts(rxn_smarts)
+
+
+def is_reaction_ok(rxn: Union[str, rdChemReactions.ChemicalReaction]):
+    """Check is the reaction is synthetically valid"""
+    if isinstance(rxn, str):
+        rxn = from_smarts(rxn_smarts=rxn)
+    return rdChemReactions.SanitizeRxn(rxn) == rdChemReactions.SanitizeFlags.SANITIZE_NONE
+
+
+def compute_reaction_product(
+    out, single_output=True, rm_attach: bool = False, as_smiles: bool = False, sanitize: bool = True
+):
+    """Compute the products of a reaction"""
+    out = [x[0] for x in out]
```

Comment:
this only consider the `first product` maybe add warning or rename the function accordingly.

Reply:
I havn't seen a good example of multiple products in the nested tuple. Would flatten the nested tuple make sense here?

Reply:
No, because there are cases where the reaction is `A.B -> C.D`. I think we should have general purpose function here and maybe have another one that takes the product at position X. X being an argument.